mvlearn/decomposition/groupica.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Authors: Pierre Ablin, Hugo Richard
+
+import numpy as np
+from picard import picard
+import scipy.linalg as linalg
+from sklearn.decomposition import fastica
+
+
+from ..utils.utils import check_Xs
+from .base import BaseEstimator
+from .grouppca import GroupPCA
+
+
+class GroupICA(BaseEstimator):
+    r"""
+    Group Independent component analysis.
+    As an optional preprocessing, each dataset in `Xs` is reduced with
```

Comment:
To me, "optional" is confusing here, because it seems to imply that this preprocessing is optional within this algorithm. It's not though, right? GroupICA necessarily includes GroupPCA? I think it would be more clear to say something like "As a preprocessing step before Independent Component Analysis, each dataset in `Xs`... "
Thoughts?

Reply:
This should indeed be clearer. The optional preprocessing step consists in reducing each dataset individually, with an individual PCA. Then, the stacking, PCA and ICA are core to the group ICA and not optional preprocessing. I will make it clearer.